src/main/java/io/qameta/allure/AllureUtils.java

```diff
@@ -109,13 +111,14 @@ private static void write(Object object, String fileName, Path outputDirectory)
         }
     }
 
-    public static String getAttachmentExtension(byte[] data) {
+    private static String getAttachmentType(byte[] data) {
         try {
             String type = getDefaultMimeTypes().detect(new ByteArrayInputStream(data), new Metadata()).toString();
-            return getExtensionByMimeType(type);
+            LOGGER.info("Detected {} mime type from data", type);
+            return type;
         } catch (IOException e) {
-            LOGGER.warn("Cannot recognize any mime type for attachment data");
-            return "";
+            LOGGER.warn("Cannot recognize any mime type for attachment's data: {}, assume 'text/plain'", e);
+            return "text/plain";
```

Comment:
Can be very dangerous if the attachment big binary file (browser will try to show it and can crash)